### PR TITLE
Feat: added iterativeFactorial

### PR DESCRIPTION
**Pull request by Renjian-buchai**
#### Description of Change
Iterative implementation of factorial. 

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: math/iterative_factorial.cpp
Comment (realstealthninja) [inline math/iterative_factorial.cpp:44]: this might be a triple slash ***not sure***

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: math/iterative_factorial.cpp
Comment (Renjian-buchai) [reply math/iterative_factorial.cpp:44]: Based on what I observed, triple slashes are for documentation while double slashes are just explanations on why it's there. Since most other algorithms use it in this way, I _think_ double slashes should be used here.

**Review by realstealthninja — CHANGES_REQUESTED** [review]
just a bit of changes to fit to contribution guidelines

File: math/iterative_factorial.cpp
Comment (realstealthninja) [reply math/iterative_factorial.cpp:44]: makes sense

File: math/iterative_factorial.cpp
Comment (realstealthninja) [inline math/iterative_factorial.cpp:11]: please use the [one](https://github.com/TheAlgorithms/C-Plus-Plus/pull/2453/commits/4ce13494136d6e60fae4f8c153bf44fd68feb86c#r1181082143) I've commented as this does not render on doxygen
```suggestion
 * \f[n! = n\times(n-1)\times(n-2)\times(n-3)\times\ldots\times3\times2\times1
 *       = n\times(n-1)!\f]
 * for example:
 * \f$4! = 4\times3! = 4\times3\times2\times1 = 24\f$
```

File: math/iterative_factorial.cpp
Comment (realstealthninja) [inline math/iterative_factorial.cpp:29]:
```suggestion
#include <cassert>    /// for assert
#include <exception>  /// for std::invalid_argument
#include <iostream>   /// for std::cout
```
triple slashes for includes!

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by realstealthninja — DISMISSED** [review]
Looks good to me!
wait for a review from @Panquesito7 
thank you for your contributions as they are greatly appreciated :heart:

File: math/iterative_factorial.cpp
Comment (Panquesito7) [inline math/iterative_factorial.cpp:8]:
```suggestion
```

File: math/iterative_factorial.cpp
Comment (Panquesito7) [inline math/iterative_factorial.cpp:3]:
```suggestion
 * @brief Iterative implementation of [Factorial](https://en.wikipedia.org/wiki/Factorial)
```

File: math/iterative_factorial.cpp
Comment (Panquesito7) [inline math/iterative_factorial.cpp:8]:
```suggestion
 * @details Calculates factorial iteratively.
```

File: math/iterative_factorial.cpp
Comment (Panquesito7) [inline math/iterative_factorial.cpp:47]: Looks like `new` crashes the whole program.
Without it, the program exits successfully.
```suggestion
        throw std::invalid_argument("Maximum n value is 20");
```

File: math/iterative_factorial.cpp
Comment (Panquesito7) [inline math/iterative_factorial.cpp:95]:
```suggestion
                 "Expected output: 2432902008176640000\n";
```

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

File: math/iterative_factorial.cpp
Comment (tjgurwara99) [inline math/iterative_factorial.cpp:56]: Minor nit (non-blocking): It could just as easily be 
```suggestion
    while (n > 1) {
        accumulator *= n;
        --n;
    }
```
since multiplying by 1 has no effect 😄

**Review by tjgurwara99 — DISMISSED** [review]

**Review by realstealthninja — APPROVED** [review]

File: math/iterative_factorial.cpp
Comment (realstealthninja) [inline math/iterative_factorial.cpp:105]: this line does not execute 
`terminate called after throwing an instance of 'std::invalid_argument*'
Aborted`

**Review by realstealthninja — COMMENTED** [review]
exception handling seems sketchy

**Review by realstealthninja — DISMISSED** [review]
Nicely done! thank you for your contribution and i expect to see more such amazing works from you! :heart:

File: math/iterative_factorial.cpp
Comment (realstealthninja) [inline math/iterative_factorial.cpp:108]:
```suggestion
                  << "\n";
```

**Review by realstealthninja — COMMENTED** [review]
nitpick:

**Review by Panquesito7 — APPROVED** [review]

**Review by realstealthninja — APPROVED** [review]
Absolutely wonderful! thank you so much for your contributions! :heart:

**Pull request merged**
